src/Documentation/tutorials_and_samples/tutorial_1_helloworld.md

```diff
@@ -0,0 +1,377 @@
+
+        private static async Task<IClusterClient> StartClientWithRetries()
+        {
+            attempt = 0;
+            IClusterClient client;
+            client = new ClientBuilder()
+                .UseLocalhostClustering()
+                .Configure<ClusterOptions>(options =>
+                {
+                    options.ClusterId = "dev";
+                    options.ServiceId = "HelloWorldApp";
+                })
+                .ConfigureLogging(logging => logging.AddConsole())
+                .Build();
+
+            await client.Connect(RetryFilter);
+            Console.WriteLine("Client successfully connect to silo host");
+            return client;
+        }
+
+        private static async Task<bool> RetryFilter(Exception exception)
+        {
+            if (exception.GetType() != typeof(SiloUnavailableException))
+            {
+                Console.WriteLine($"Cluster client failed to connect to cluster with unexpected error.  Exception: {exception}");
+                return false;
+            }
+            attempt++;
+            Console.WriteLine($"Cluster client attempt {attempt} of {initializeAttemptsBeforeFailing} failed to connect to cluster.  Exception: {exception}");
+            if (attempt > initializeAttemptsBeforeFailing)
+            {
+                return false;
+            }
+            await Task.Delay(TimeSpan.FromSeconds(4));
+            return true;
+        }
+
+        private static async Task DoClientWork(IClusterClient client)
+        {
+            // example of calling grains from the initialized client
+            var friend = client.GetGrain<IHello>(0);
+            var response = await friend.SayHello("Good morning, my friend!");
+            Console.WriteLine("\n\n{0}\n\n", response);
+        }
+    }
+}
+
+
+```
+
+### HelloWorld.Interfaces - IHello.cs
+
+```csharp
+using System.Threading.Tasks;
+
+namespace HelloWorld.Interfaces
+{
+    /// <summary>
+    /// Orleans grain communication interface IHello
+    /// </summary>
+    public interface IHello : Orleans.IGrainWithIntegerKey
+    {
+        Task<string> SayHello(string greeting);
+    }
+}
+
+
+```
+
+### HelloWorld.Grain - HelloGrain.cs
+
+```csharp
+using HelloWorld.Interfaces;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace HelloWorld.Grains
+{
+    /// <summary>
+    /// Orleans grain implementation class HelloGrain.
+    /// </summary>
+    public class HelloGrain : Orleans.Grain, IHello
+    {
+        private readonly ILogger logger;
+
+        public HelloGrain(ILogger<HelloGrain> logger)
+        {
+            this.logger = logger;
+        }  
+
+        Task<string> IHello.SayHello(string greeting)
+        {
+            logger.LogInformation($"SayHello message received: greeting = '{greeting}'");
+            return Task.FromResult($"You said: '{greeting}', I say: Hello!");
+        }
+    }
+}
+
+```
+
+## Step Five: Deploy the Solution
+
+### Using Visual Studio
```

Comment:
these are duplicated informaton with HelloWorldSample/ReadMe.md. I think we can move those to the HelloWorldSample/ReadMe.md, for two reasons IMO 
- so people who are just looking at the sample code will know how to run it. Not everyone read docs. Keep it with sample code make the sample more complete 
- to keep toturial precise and to the point. I think the goal for this tutorial is, pairing with the sample code, giving people a intro to orleans core concepts, which is grain, silo and client